src/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -22,7 +25,10 @@ interface Props {
 
 export default function ColumnHeader({ columnIndex, columnName, dataReady, direction, onClick, orderByIndex, orderBySize, ariaColIndex, ariaRowIndex, className, children }: Props) {
   const ref = useRef<HTMLTableCellElement>(null)
+  const buttonRef = useRef<HTMLDivElement>(null)
   const { tabIndex, navigateToCell } = useCellNavigation({ ref, ariaColIndex, ariaRowIndex })
+  const { isOpen, position, menuId, handleToggle, handleMenuClick } =
+    useColumnMenu(columnIndex, ref, navigateToCell)
```

Comment:
Add useColumnMenu hook to separate concerns

ColumnHeader was handling both column display and menu logic. This hook isolates menu state management, making the component easier to understand and test.